### PR TITLE
feat: support for case-insensitive matches

### LINKED ISSUE #12: Fix `URLPatternResult#inputs` behaviour
There is an inconsistency between spec and tests and Chromium. Needs a spec decision https://github.com/WICG/urlpattern/issues/133.

### PR DESCRIPTION
**Pull request by crowlKats**
Implements WICG/urlpattern#168
Closes #12

### CODE REVIEW

**Comment by crowlKats** [comment]
> generate_matcher is going to significantly suffer here. It will be unable to generate matchers for case insensitive parts

thats a problem, but i dont think there is much we can do (I'd be happily proven wrong)

**Comment by vazexqi** [comment]
@crowlKats @lucacasonato - Any chance that this might land in Deno? According to https://developer.mozilla.org/en-US/docs/Web/API/URLPattern#browser_compatibility Deno has everything else implemented according to the spec except for the `ignoreCase` feature.

File: src/lib.rs [hunk 1]
```diff
@@ -491,6 +513,7 @@ impl<R: RegExp> UrlPattern<R> {
     let hostname_exec_result = self.hostname.matcher.matches(&input.hostname);
     let port_exec_result = self.port.matcher.matches(&input.port);
     let pathname_exec_result = self.pathname.matcher.matches(&input.pathname);
+    dbg!(&pathname_exec_result, &input.pathname);
```
Comment (lucacasonato) [inline src/lib.rs:516]: Remove

File: src/testdata/urlpatterntestdata.json [hunk 2]
```diff
@@ -2787,7 +2782,7 @@
     }
   },
   {
-    "skip": "ingoreCase is not yet implemented",
+    "skip": "this does not error. same happens in Chrome as well.",
```
Comment (crowlKats) [inline src/testdata/urlpatterntestdata.json:2785]: This is a bit strange, but not sure what is up with it.

Comment (lucacasonato) [reply src/testdata/urlpatterntestdata.json:2785]: The test passes on Chrome in WPT: 
image

**Review by lucacasonato — APPROVED** [review]
LGTM

**Pull request merged**